src/auth/User.h

```diff
@@ -48,7 +51,8 @@ class User : public RefCountable
     Auth::Type auth_type;
     /** the config for this user */
     Auth::SchemeConfig *config;
-    dlink_list proxy_match_cache;
+    // XXX would be better served by unordered_map<void*, int> ?
+    std::list<acl_proxy_auth_match_cache> proxy_match_cache;
```

Comment:
Please add a brief description of this data member. What exactly are we caching here? What is the lifetime/expiration policy of those cached items? Most of that description can be outsourced to the typedef for the new type.

Reply:
Ok. It's going to be lightweight